dist/videojs-errors.js

```diff
@@ -140,14 +138,16 @@
 
     // PLAYER_ERR_NO_SRC
     player.on('play', function() {
-      if (player.currentSrc() === null ||
-          player.currentSrc() === undefined ||
-          player.currentSrc() === '') {
-        player.error({
-          code: -1,
-          type: 'PLAYER_ERR_NO_SRC'
-        });
-      }
+      window.setTimeout(function(){
+        if (player.currentSrc() === null ||
+            player.currentSrc() === undefined ||
+            player.currentSrc() === '') {
+          player.error({
+            code: -1,
+            type: 'PLAYER_ERR_NO_SRC'
+          });
+        }
+      }, 500);
```

Comment:
Looks like the video element does nothing no autoplay with no source: https://jsfiddle.net/gkatsev/as03tgdp/5/
Then if you add the source later, it'll play at that time: https://jsfiddle.net/gkatsev/as03tgdp/6/
(This is without calling play directly ourselves).